contracts/CourtPresaleActivate.sol

```diff
@@ -53,16 +53,21 @@ contract CourtPresaleActivate is IsContract, ApproveAndCallFallBack {
     * @param _from Address of the original caller (juror) converting and activating the tokens
     * @param _amount Amount of contribution tokens to be converted and activated
     * @param _token Address of the contribution token triggering the approve-and-call fallback
+    * @param _data If non-empty it will signal token activation in the registry
     */
-    function receiveApproval(address _from, uint256 _amount, address _token, bytes calldata) external {
+    function receiveApproval(address _from, uint256 _amount, address _token, bytes calldata _data) external {
         require(_amount > 0, ERROR_ZERO_AMOUNT);
         require(_token == address(presale.contributionToken()), ERROR_WRONG_TOKEN);
 
         // move tokens to this contract
         ERC20 token = ERC20(_token);
         require(token.safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
 
-        _buyAndActivate(_from, _amount, _token);
+        bool activate;
+        if (_data.length > 0) {
+            activate = true;
+        }
+        _buyAndActivate(_from, _amount, _token, activate);
```

Comment:
Perhaps we can split up `_buyAndActivate()` so that we can split up the action at the call site rather than in this function?

Reply:
What do you mean by "call site"? The frontend? If that's the case, I see 2 problems:
- It's being called from 3 different places, se we would have to split those 3 endpoints
- One of those endpoints is the ApproveAndCall fallback, so it doesn't come directly from the frontend.

Reply:
"Call site" as in where it's called in the contract.

You're right though, it wouldn't be ideal to push the logic for activation into three places, especially if we're using the logic to emitting events / etc.

Perhaps we can just rename `_buyAndActivate()` to `_buyAndPossiblyActivate()` (or even just `_contributeToPresale()`)?